### PR TITLE
use stringer if string tag option is used

### PR DESCRIPTION
**Pull request by kunalpowar**
Added feature to allow users to add `string` option to struct tags just like json package.

### CODE REVIEW

**Comment by kunalpowar** [comment]
I need to handle `omitempty`.

File: structs.go [hunk 1]
```diff
@@ -89,6 +99,16 @@ func (s *Struct) Map() map[string]interface{} {
...
 		}
 
+		if tagOpts.Has("string") {
+			_, ok := val.Type().MethodByName("String")
+			if !ok {
+				panic(fmt.Sprintf("field %s is not a stringer", name))
+			}
+
+			out[name] = val.MethodByName("String").Call([]reflect.Value{})[0].String()
+			continue
```
Comment (fatih) [inline structs.go]: * Can you rename from `string` to `stringer`? 
* We should not panic, we should just continue. It should be lazily loaded
* This is a better approach:
```go
if s, ok := val.Interface().(fmt.Stringer); ok {
	out[name] = s.String()
}
```
* Also put this after the last block, so it should come **after** the block:

```go
if IsStruct(val.Interface()) && !tagOpts.Has("omitnested") {
}
else {
} 
...


< Put code here> 
```

**Comment by fatih** [comment]
Hi @kunalpowar Great feature! thanks for your contribution. I made some comment, they applies to the second piece of code too. If you can fix them we can easily merge.

**Comment by kunalpowar** [comment]
Thanks for the feedback @fatih. I'll make those changes asap.

File: structs.go [hunk 2]
```diff
@@ -153,6 +171,14 @@ func (s *Struct) Values() []interface{} {
 			}
 		}
 
+		if tagOpts.Has("string") {
+			s, ok := val.Interface().(fmt.Stringer)
+			if ok {
+				t = append(t, s.String())
+			}
+			continue
+		}
```
Comment (fatih) [inline structs.go]: Can you also move this under the below if block?

Comment (kunalpowar) [inline structs.go]: Considering the possibility of an embedded struct being a Stringer, shouldn't the `string` tag check happen before the embedded struct check?

File: structs_test.go [hunk 3]
```diff
@@ -910,3 +910,89 @@ func TestNestedNilPointer(t *testing.T) {
...
+
+	s.TagName = "json"
+	m := s.Map()
+
+	if m["person"] != person.String() {
+		t.Errorf("Value for field person should be %s, got: %s", person.String(), m["person"])
+	}
+
+	vs := s.Values()
+	if vs[1] != person.String() {
```
Comment (kunalpowar) [inline structs_test.go]: If we don't check for `string` tag before embedded struct check, the stringer for `Person` will never be used.

**Comment by fatih** [comment]
This looks good @kunalpowar Can you please rebase and squash to a single commit?

**Comment by kunalpowar** [comment]
Doing it now.

**Comment by kunalpowar** [comment]
Done.

**Comment by fatih** [comment]
Thanks @kunalpowar :+1:

**Comment by kunalpowar** [comment]
Thanks @fatih :)

**Pull request merged**